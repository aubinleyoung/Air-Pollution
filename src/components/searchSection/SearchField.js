/* eslint-disable react/no-array-index-key */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import BounceLoader from 'react-spinners/BounceLoader';
import { NavLink, useParams } from 'react-router-dom';
import { color, override, baseImgSrc } from '../globalUtilities/utility';
import { displayCountryMap } from '../../redux/POLLUTION/pollution';
import { fecthCountry } from '../../redux/Country/countries';
import search from './Search.module.css';

const SearchField = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.country_name);
  const { country, loading } = countries;
  const { id } = useParams();
  const namePass = id ? id.toLowerCase() : 'europe';

  useEffect(() => {
    if (country.length === 0) {
      dispatch(fecthCountry(namePass));
    }
  }, []);

  const setDisplay = () => {
    dispatch(displayCountryMap(true));
  };

  const countryField = country.length !== 0
    ? country[0].map((field, index) => (
      <NavLink
        to={`/pollution/${field.latlag}:${field.name}:${field.code}`}
        className={search.nav}
        key={index}
        onClick={setDisplay}
      >
        <div className="border p-5">
          <div className={search.img_container}>
            <img
              src={`${baseImgSrc}/${field.code}/vector.svg`}
              alt="country code"
            />
          </div>
          <div>
            {' '}
            <span className={search.field_name}>
              {' '}
              {field.name}
              {' '}
            </span>
            {' '}
            <div>
              {' '}
              Population:
              {field.population}
            </div>
            {' '}
          </div>
        </div>
      </NavLink>
    ))
    : ' ';

  return (
    <div className="">
      <h4 className="py-5 px-5 text-white font-bold" data-testid="search">
        {' '}
        STATES BY COUNTRY
        {' '}
      </h4>
      <div className="grid grid-cols-2">{countryField}</div>
      <BounceLoader color={color} css={override} loading={loading} size={60} />
    </div>
  );
};

export default SearchField;
