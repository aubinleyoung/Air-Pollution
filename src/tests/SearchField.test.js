/* eslint-disable comma-dangle */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import SearchField from '../components/searchSection/SearchField';
import '@testing-library/jest-dom';

afterEach(cleanup);
describe('Hero Component', () => {
  test('Render Pollution', () => {
    render(
      <Router>
        <Provider store={store}>
          <SearchField />
        </Provider>
      </Router>
    );

    const pollutionElement = screen.getByTestId('search');

    expect(pollutionElement).toHaveTextContent('STATES BY COUNTRY');
  });

  test('Snapshot Pollution', () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <SearchField />
          </Provider>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
