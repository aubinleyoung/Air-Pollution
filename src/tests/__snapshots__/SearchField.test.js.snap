// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component Snapshot Pollution 1`] = `
<div
  className=""
>
  <h4
    className="py-5 px-5 text-white font-bold"
    data-testid="search"
  >
     
    STATES BY COUNTRY
     
  </h4>
  <div
    className="grid grid-cols-2"
  >
     
  </div>
  <span
    css={
      Object {
        "map": undefined,
        "name": "13gkr5z",
        "next": undefined,
        "styles": "
  display: block;
  margin: 4em auto;
",
        "toString": [Function],
      }
    }
    style={
      Object {
        "display": "inherit",
        "height": "60px",
        "position": "relative",
        "width": "60px",
      }
    }
  >
    <span
      style={
        Object {
          "animation": "react-spinners-BounceLoader-bounce 2.1s 1s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#56deff",
          "borderRadius": "100%",
          "height": "60px",
          "left": 0,
          "opacity": 0.6,
          "position": "absolute",
          "top": 0,
          "width": "60px",
        }
      }
    />
    <span
      style={
        Object {
          "animation": "react-spinners-BounceLoader-bounce 2.1s 0s infinite ease-in-out",
          "animationFillMode": "both",
          "backgroundColor": "#56deff",
          "borderRadius": "100%",
          "height": "60px",
          "left": 0,
          "opacity": 0.6,
          "position": "absolute",
          "top": 0,
          "width": "60px",
        }
      }
    />
  </span>
</div>
`;
